packages/edit-widgets/src/store/batchSupport.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { STATE_ERROR } from '@wordpress/batch-processing';
+import { dispatch } from '@wordpress/data';
+
+function shoehornBatchSupport() {
+	apiFetch.use( ( options, next ) => {
+		if (
+			! [ 'POST', 'PUT', 'PATCH', 'DELETE' ].includes( options.method ) ||
+			! isWidgetsEndpoint( options.path )
+		) {
+			return next( options );
+		}
+		return addToBatch( options );
+	} );
+
+	dispatch( 'core/batch-processing' ).registerProcessor(
+		'WIDGETS_API_FETCH',
+		batchProcessor
+	);
+}
+
+function isWidgetsEndpoint( path ) {
+	// This should be more sophisticated in reality:
+	return (
+		path.indexOf( '/widgets' ) !== -1 || path.indexOf( '/sidebars' ) !== -1
+	);
+}
+
+// setTimeout is a hack to ensure batch-processing store is available for dispatching
+setTimeout( shoehornBatchSupport );
+
+const addToBatch = ( request ) => {
+	return dispatch( 'core/batch-processing' ).enqueueItemAndAutocommit(
+		'WIDGETS_API_FETCH',
+		'default',
+		request
+	);
```

Comment:
Alternatively this could call something like `enqueueItemAndWaitForResults` and get a promise that would be resolved once `processBatch` is called explicitly.